pkg/cpio/cpio.go

```diff
@@ -140,17 +138,78 @@ func (w Writer) Concat(r Reader, transform func(Record) Record) error {
 	}
 }
 
-// MakeReproducible changes any fields in a Record such that
-// if we run cpio again, with the same files presented to it
-// in the same order, and those files have unchanged contents,
-// the cpio file it produces will be bit-for-bit
-// identical. This is an essential property for firmware-embedded
-// payloads.
-func MakeReproducible(file Record) Record {
-	file.MTime = 0
-	return file
+// Archive is an in-memory list of files.
+type Archive struct {
+	// Files is a map of relative archive path -> record.
+	Files map[string]Record
+
+	// Order is a list of relative archive paths and represents the order
+	// in which Files were added.
+	Order []string
+}
+
+// Add adds a record to the archive.
+func (a *Archive) Add(r Record) {
+	a.Files[r.Name] = r
```

Comment:
It would be nice to have an error here in case two records with the same name are added.